### PR TITLE
Documentation site

### PR DESCRIPTION
**Pull request by Menduist**
Follow up of #716

Using material for mkdocs to generate a nicer website:
https://status-im.github.io/nim-libp2p/docs/

(Note that the "reference" link will work once this hits master)

### CODE REVIEW

File: .github/workflows/doc.yml [hunk 1]
```diff
@@ -61,3 +61,39 @@ jobs:
           git config --global user.name = "${{ github.actor }}"
           git commit -a -m "update docs for ${GITHUB_REF##*/}"
           git push origin gh-pages
+
+  update_site:
+    if: github.ref == 'refs/heads/unstable'
```
Comment (Menduist) [inline .github/workflows/doc.yml:66]:
```suggestion
    if: github.ref == 'refs/heads/master'
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# nim-libp2p documentation
+
+Welcome to the nim-libp2p documentation!
+
+Here, you'll find [tutorials](tutorial_1_connect.md) to help you started, as well as [examples](directchat.nim) and
```
Comment (Menduist) [inline examples/README.md:5]:
```suggestion
Here, you'll find [tutorials](tutorial_1_connect.md) to help you get started, as well as [examples](directchat.nim) and
```

**Review by lchenut — APPROVED** [review]

**Comment by arnetheduck** [comment]
Similar to the documentation in nimbus, you need to lock down the version of material/mkdocs installed or different people will generate different sites, and the CI might break - see https://github.com/status-im/nimbus-eth2/blob/d64c17ffc3525101261ad21464f5fe53f6c3f868/docs/Makefile#L5

**Comment by arnetheduck** [comment]
also, keeping docs under a "docs" folder is no a bad idea for finding them ;)

**Pull request merged**